--- rlinetd-0.6/configure.ac.orig	2006-03-19 16:29:48.000000000 +0100
+++ rlinetd-0.6/configure.ac	2006-03-20 10:12:05.089062000 +0100
@@ -42,7 +42,7 @@
 libnsl=yes
 AC_ARG_WITH(libnsl,dnl
   --with-libnsl           link with the nsl library,
-	libwnsl=$withval)
+	libnsl=$withval)
 if test x"$libnsl" != x"no"; then
   AC_CHECK_LIB(nsl, yp_get_default_domain, libsrlinetd="${libsrlinetd} -lnsl"; LIBS="$LIBS -lnsl")
 fi  
