--- rlinetd-0.5.1/assemble.c.orig	2004-08-03 19:21:44.946639000 +0000
+++ rlinetd-0.5.1/assemble.c	2004-08-03 19:22:59.939802598 +0000
@@ -174,7 +174,7 @@
 			if(ret->elems[j] == b->elems[i])
 				goto found;
 		array_append(ret, b->elems[i]);
-found:
+found: ;
 	}
 	return ret;
 }
--- rlinetd-0.5.1/grammar.y.orig	2004-08-03 19:21:44.947616000 +0000
+++ rlinetd-0.5.1/grammar.y	2004-08-03 19:23:36.480817776 +0000
@@ -1145,7 +1145,7 @@
 					goto out;
 				}
 				succeeded++;
-			out:
+			out: ;
 #ifdef HAVE_GETADDRINFO
 			} while((ai = ai->ai_next));
 #endif
